### PR TITLE
files_external: Add OpenAPI spec

### PR DESCRIPTION
**Pull request by provokateurin**
## Summary

From https://github.com/nextcloud/server/pull/36666. Adds the necessary annotations and descriptions.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files_external/openapi.json [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+                    }
+                ],
+                "parameters": [
+                    {
+                        "name": "OCS-APIRequest",
+                        "in": "header",
+                        "required": true,
+                        "schema": {
+                            "type": "string",
+                            "default": "true"
```
Comment (bigcat88) [inline apps/files_external/openapi.json:120]: Is this is real by default `true` and present?

**Comment by provokateurin** [comment]
@artonge @icewind1991 Can you help me with the CI failure? I don't see why my changes would cause any failure since they only add annotations which are completely ignored at runtime, thus no behavioral changes were made afaict

**Review by artonge — APPROVED** [review]
I think some of those tests are triggered only when some specific files are edited.
They might be failing for a long time :see_no_evil: .
As your changes are mostly comments, I think we can merge without further concern.
If @icewind1991 agrees, feel free to merge :)

**Comment by provokateurin** [comment]
Yeah I already imagined something like that. There is no way the comments can break something

**Review by bigcat88 — APPROVED** [review]

**Comment by icewind1991** [comment]
S3 external test fix is here: https://github.com/nextcloud/server/pull/39367

**Pull request merged**